### PR TITLE
COMMERCE-10603 fetch category by ERC first

### PR DESCRIPTION
**Pull request by jaydawu**
https://issues.liferay.com/browse/COMMERCE-10603
Resent from https://github.com/liferay-commerce/liferay-portal/pull/3731

### CODE REVIEW

**Comment by jaydawu** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
ci:test:relevant

**Comment by jaydawu** [comment]
ci:test:commerce-functional

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProducts.testcase [hunk 1]
```diff
@@ -1985,6 +1985,86 @@ definition {
...
+
+		task ("And the category created before is imported in the virtual instance") {
+			Category.openCategoriesAdmin(
+				baseURL = "http://www.able.com:8080",
+				siteURLKey = "global");
+
+			LAR.importPortlet(larFileName = "Category_Topic_Cat.portlet.lar");
+		}
+
+		task ("And a product that imported that has the same ERC category via JSON is added in the virtual instance") {
```
Comment (lorenzolapescara) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProducts.testcase:2046]:
```suggestion
		task ("And a product with the imported category is added to the virtual instance via JSON") {
```

**Review by lorenzolapescara — CHANGES_REQUESTED** [review]

**Comment by jaydawu** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
ci:test:commerce-functional

**Comment by jaydawu** [comment]
ci:test:relevant

**Comment by jaydawu** [comment]
ci:test:relevant

**Comment by jaydawu** [comment]
ci:test:commerce-functional

**Comment by jaydawu** [comment]
ci:test:relevant

**Comment by jaydawu** [comment]
@andrea-ale-sbarra could you please fix the conflict in ProductResourceImpl.java file that you used? Thanks in advance.

**Comment by andrea-ale-sbarra** [comment]
conflicts fixed here https://github.com/liferay-commerce/liferay-portal/pull/3748

**Pull request closed without merging**